docs/sql-ref-syntax-qry-select-transform.md

```diff
@@ -0,0 +1,244 @@
+---
+layout: global
+title: TRANSFORM
+displayTitle: TRANSFORM
+license: |
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+ 
+     http://www.apache.org/licenses/LICENSE-2.0
+ 
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+---
+
+### Description
+
+The `TRANSFORM` clause is used to specify a Hive-style transform query specification 
+to transform the inputs by running a specified script. Users can plug in their own custom 
+scripts in the data stream by using features natively supported in the Spark SQL.
+
+### Syntax
+
+```sql
+SELECT TRANSFORM ( named_expression [ , ... ] )
```

Comment:
`named_expression` -> `expression`

Reply:
> `named_expression` -> `expression`

Since this doc for spark-3.0/3.1/master so here we don't change this. Will rase a follow up pr to update this to current master's behavior

Reply:
If we agree supporting alias is a mistake, we shouldn't document it even in 3.0/3.1

Reply:
> If we agree supporting alias is a mistake, we shouldn't document it even in 3.0/3.1

reasonable, updated.